HIP/hip-xxx.md

```diff
@@ -0,0 +1,191 @@
+
+## Specification
+
+In order to enable trace calls, we should meet some requirements on the Mirror Archive Node side:
+
+- Support EVM versioning, so that we use the appropriate EVM version for a specific historical block.
+- Use full mirror node, keeping state from the genesis block and configured with enabled ingest of `CONTRACT_BYTECODE` , `CONTRACT_STATE_CHANGE` and `CONTRACT_ACTION` sidecar types.
+- Re-execute historical transactions using the information for state values read from the downloaded `CONTRACT_STATE_CHANGE` sidecar record files.
+
+### JSON-RPC Relay Trace API
+
+The JSON-RPC Relay will be enhanced with a `debug_traceTransaction` endpoint. The endpoint has one required parameter - `transactionHash` - and also accepts a list of properties - a `tracer`, and a list of configuration `flags`.
+
+Example payload:
+
+```jsx
+{
+	"jsonrpc": "2.0",
+	"method": "debug_traceTransaction",
+	"params": [
+      // The transaction hash.
+	  "0x123...", 
+	  {
+		// The tracer to use. Accepted values: `callTracer` and `opcodeLogger`. Optional field. Default value is `callTracer`.
+		tracer: "callTracer"
+		// Options for `opcodeLogger`. Ignored for `callTracer`. Disabling some of the metrics collected will reduce the output.
+	    disableMemory: true,
+	    disableStack: true,
+	    disableStorage: true,
+	  },
+	],
+	"id": 0
+}
+```
+
+### Mirror node Trace REST API
+
+The JSON-RPC Relay will in turn call the Mirror Node REST API. Depending on the `tracer` that was provided as an input parameter to the `debug_traceTransaction` RPC endpoint, one of two Mirror node endpoints will be called:
+1. For `debug_traceTransaction` called with `callTracer`, the Mirror Node `/api/v1/contracts/results/{transactionIdOrHash}/actions` endpoint will be called.
+2. For `debug_traceTransaction` called with `opcodeLogger`, the Mirror Node `/api/v1/contracts/results/{transactionIdOrHash}/opcodes` endpoint will be called.
+
+The `/actions` endpoint is already implemented. 
+
+This section will specify the `/opcodes` endpoint.
+
+#### `/api/v1/contracts/results/{transactionIdOrHash}/opcodes`
+
+**Input parameters:** The endpoint will accept three optional parameters:
+
+```
+// Boolean flag controlling whether to return stack information. Defaults to `true`.
+
+stack: <bool> 
+
+// Boolean flag controlling whether to return memory information. Defaults to `false`.
+
+memory: <bool> 
+
+// Boolean flag controlling whether to return storage information. Defaults to `false`.
+
+storage: <bool>
+```
+
+Disabling stack, memory and storage traces will significantly reduce the size of the opcode related data. This would be helpful for reducing the network bandwidth. 
+
+**Output**: detailed information for each executed op code.
+Example output:
+```
+{
+    "jsonrpc": "2.0",
+    "id": 0,
+    "result": {
+        "gas": 52139,
+        "failed": false,
+        "returnValue": "0000000000000000000000000000000000000000000000000000000000000001",
```

Comment:
We should probably add 0x prefix to hex values like the rest of the API

Reply:
I updated the examples for the REST API to include prefixes.